lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -267,6 +268,204 @@ namespace Js
         return promiseCapability->GetPromise();
     }
 
+    Var JavascriptPromise::EntryAny(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        AUTO_TAG_NATIVE_LIBRARY_ENTRY(function, callInfo, _u("Promise.any"));
+
+        // 1. Let C be the this value.
+        Var C = args[0];
+        RecyclableObject* constructor = VarTo<RecyclableObject>(C);
+
+        // 2. Let promiseCapability be ? NewPromiseCapability(C).
+        JavascriptPromiseCapability* promiseCapability = NewPromiseCapability(C, scriptContext);
+
```

Comment:
There needs to be a type check before the VarTo and it can then be an UnsafeVarTo. NewPromiseCapability does the typecheck we need so can switch this around:

```c
JavascriptPromiseCapability* promiseCapability = NewPromiseCapability(C, scriptContext);
RecyclableObject* constructor = UnsafeVarTo<RecyclableObject>(C);
```

With the current version it will abort (hard crash) if you do:
```js
Promise.any.call(5)
```

(Sorry I missed this one before)